core/src/main/java/org/apache/iceberg/io/RollingFileWriter.java

```diff
@@ -127,9 +130,8 @@ private void closeCurrentWriter() {
       if (currentFileRows == 0L) {
         try {
           io.deleteFile(currentFile.encryptingOutputFile());
-        } catch (UncheckedIOException e) {
-          // the file may not have been created, and it isn't worth failing the job to clean up,
-          // skip deleting
+        } catch (Exception e) {
```

Comment:
Sorry for the delay on review, I think it makes sense to widen the exception for this case but if we really want to solve this for all cases I think we would want to catch `Throwable` instead of `Exception`.

Reply:
Sorry for my delay as well. I've changed `Exception` to `Throwable`